File: app/src/main/java/com/testapp/testapp/presenter/adapter/CommonRecyclerViewAdapter.java
package com.testapp.testapp.presenter.adapter;

import android.support.v7.widget.RecyclerView;

import java.util.List;

/**
 * Created by troll on 07.11.2017.
 */

public abstract class CommonRecyclerViewAdapter<T, VH extends RecyclerView.ViewHolder> extends RecyclerView.Adapter<VH>{

    public abstract void addItems(List<T> items);
    public abstract void clearItems();
}

File: app/src/main/java/com/testapp/testapp/kotlin/presenter/adapter/CommonRecyclerViewAdapter.kt
package com.testapp.testapp.kotlin.presenter.adapter

import android.support.v7.widget.RecyclerView

/**
 * Created by dmitriysmishnyi on 28.08.18.
 */
abstract class CommonRecyclerViewAdapter<T, VH : RecyclerView.ViewHolder> : RecyclerView.Adapter<VH>() {

    abstract fun addItems(items: List<T>)
    abstract fun clearItems()
}